crates/op-rbuilder/src/flashtestations/service.rs

```diff
@@ -0,0 +1,239 @@
+        if (args.debug) {
+            info!(target: "flashtestations", "running in debug mode - will use HTTP service for quotes");
+        }
+        let attestation_provider = get_attestation_provider(AttestationConfig {
+            debug: args.debug,
+            debug_url: args.debug_url,
+        });
+        let onchain_provider = OnchainProvider::new(
+            signer,
+            args.rpc_url,
+            args.registry_address,
+            args.builder_policy_address,
+        );
+        
+        // Step 1: Generate Ethereum keypair
+        let (private_key, public_key, address) = generate_ethereum_keypair();
+
+        // Step 2: Prepare report data with public key (64 bytes, no 0x04 prefix)
+        let mut report_data = [0u8; 64];
+        let pubkey_uncompressed = public_key.serialize_uncompressed();
+        report_data.copy_from_slice(&pubkey_uncompressed[1..65]); // Skip 0x04 prefix
+
+        // Step 3: Request TDX attestation
+        info!(target: "flashtestations", "requesting TDX attestation");
+        let attestation = attestation_provider.get_attestation(report_data)?;
+
+        // Step 4: Submit report onchain with the builder key
+        onchain_provider.register_tee_service(attestation)?;
+
+        Ok(Self {
+            attestation_provider,
+            onchain_provider,
+            tee_service_signer: Signer::try_from_secret(private_key)?,
+        })
+    }
+
+    pub fn signed_builder_proof_tx(&self, block: Block) -> eyre::Result<OpTransactionSigned> {
+        let block_content_hash = compute_block_content_hash(&block);
+        let tx_builder = self.onchain_provider.verify_block_builder_proof(block_content_hash)?;
+        // TODO: sign the transaction with TEE service key
+        // let tx = self.tee_service_signer.sign_tx(tx_builder)?;
+        Ok(tx)
+    }
+}
+
+/// Computes the block content hash according to the formula:
+/// keccak256(abi.encode(parentHash, blockNumber, timestamp, transactionHashes))
+fn compute_block_content_hash(block: &BlockInfo, transactions: &[TxEnvelope]) -> B256 {
+    // Step 1: Create ordered list of transaction hashes
+    let transaction_hashes: Vec<B256> = transactions
+        .iter()
+        .map(|tx| {
+            // RLP encode the transaction and hash it
+            let mut encoded = Vec::new();
+            tx.encode(&mut encoded);
+            keccak256(&encoded)
+        })
+        .collect();
+
+    // Step 2: ABI encode the data
+    // ABI encoding format:
+    // - parentHash: bytes32 (32 bytes)
+    // - number: uint256 (32 bytes)
+    // - timestamp: uint256 (32 bytes)
+    // - transactionHashes: bytes32[] (dynamic array)
+
+    let encoded = abi_encode_block_data(
+        block.parent_hash,
+        block.number,
+        block.timestamp,
+        &transaction_hashes,
+    );
+
+    // Step 3: Hash the ABI encoded data
+    keccak256(&encoded)
+}
+
+/// ABI encodes: (bytes32, uint256, uint256, bytes32[])
+fn abi_encode_block_data(
```

Comment:
You can do something like this to encode an object https://docs.rs/alloy-sol-types/latest/alloy_sol_types/#solstruct